storage/url/url.go

```diff
@@ -45,10 +44,26 @@ type URL struct {
 	relativePath string
 	filter       string
 	filterRegex  *regexp.Regexp
+	mode         URLMode
+}
+
+type URLMode int
```

Comment:
using a bool field instead of URLMode type will simplify checks below:
```go
    var urlMode url.URLMode
    if c.raw {
        urlMode = 1
    }
    srcurl, err := url.New(c.src, url.WithMode(urlMode))
```

Above sample can be written as this if we use a bool field.
```go
srcurl, err := url.New(c.src, url.WithMode(c.raw))
```

Reply:
changed as you suggested.